sycl/doc/Assert.md

```diff
@@ -347,6 +342,19 @@ fail.
 
 ### Raising assert failure flag and reading it on host
 
+In DPCPP headers one can see if assert is enabled with status of `NDEBUG` macro
+with `#ifdef`'s. Though, in order to support for multi translation unit use-case
+it's not allowed to rely on definition of `NDEBUG` macro.
+
+*Note: Multi translation unit use-case here is the one with `SYCL_EXTERNAL`
+function compiled with assertions enabled and used in a kernel  but the kernel
+is compiled with assertions disabled.*
+
+There're two commands used for reading assert failure flag: copy kernel and
+checker host task. The copy kernel will copy `AssertHappenedMem` to host and
+host-task will check the `Flag` value and `abort()` as needed. The kernel and
+host task are enqueued when `NDEBUG` macro isn't defined.
```

Comment:
I think the only decision factor to enqueue or not in the runtime should be presence of the
"uses asserts" property, so:
```suggestion
host-task will check the `Flag` value and `abort()` as needed. The kernel and
host task are enqueued together with a kernel only when the corresponding device
binary image property for this kernel tells that it may use (maybe indirectly) the `assert`
in its code.
```